File: tf1testech.client/src/models/LeavePeriodDemandDTO.ts
import { LeavePeriodStatus } from "./enums/LeavePeriodStatus";
import { LeavePeriodType } from "./enums/LeavePeriodType";

class LeavePeriodDemandDTO {
    comment: string;
    managerComment: string;
    startDate: string;
    endDate: string;
    type: LeavePeriodType;
    status: LeavePeriodStatus;

    constructor(comment: string, managerComment:string, startDate: string, endDate: string, type: LeavePeriodType, status: LeavePeriodStatus){
        this.comment = comment;
        this.managerComment = managerComment;
        this.startDate = startDate;
        this.endDate = endDate;
        this.type = type;
        this.status = status;
    }
}

export default LeavePeriodDemandDTO;

File: Domain/ValueObjects/LeavePeriodDemandDTO.cs
using Domain.Entities;
using Domain.Entities.Enums;

namespace Domain.ValueObjects
{
    public record LeavePeriodDemandDTO(string Comment,
        string ManagerComment,
        DateTime StartDate,
        DateTime EndDate,
        LeavePeriodType Type,
        LeavePeriodStatus Status)
    {
        public LeavePeriodDemand ToDomain()
        {
            return new LeavePeriodDemand(Status) 
            {
                Comment = Comment,
                ManagerComment = ManagerComment,
                StartDate = StartDate,
                EndDate = EndDate,
                Type = Type,
            };
        }
    }
}
